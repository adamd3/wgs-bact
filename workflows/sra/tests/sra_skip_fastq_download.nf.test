nextflow_workflow {

    name "Test workflow: sra/main.nf"
    script "../main.nf"
    workflow "SRA"
    tag "workflows"
    tag "workflows_sra"
    tag "multiqc_mappings_config"
    tag "sra_skip_fastq_download"

    test("Parameters: --skip_fastq_download") {

        when {
            workflow {
                """
                input[0] = Channel.from('ERR1160846', 'GSE214215', 'SRR12848126')
                """
            }
            params {
                outdir = "results"
                skip_fastq_download = true
            }
        }

        then {
            assert workflow.success

            with(workflow.out.samplesheet) {
                assert path(get(0)).readLines().size() == 5
                assert path(get(0)).readLines()*.split(',').collect { it[0] } == ['"sample"', '"SRX17709227"', '"SRX17709228"', '"ERX1234253"', '"SRX9315476"']
                assert path(get(0)).text.contains('Illumina HiSeq 4000')
            }

            with(workflow.out.mappings) {
                assert path(get(0)).readLines().size() == 5
                assert path(get(0)).readLines()*.split(',').collect { it[0] } == ['"sample"', '"SRX17709227"', '"SRX17709228"', '"ERX1234253"', '"SRX9315476"']
                assert path(get(0)).text.contains('Illumina HiSeq 4000')
            }

            with(workflow.out.sample_mappings) {
                assert path(get(0)[0]).md5 == "6d4c3e5137704358474330207f5f2b5c"
            }

            with(workflow.out.sra_metadata) {
                assert get(0).get('experiment_accession') == 'ERX1234253'
                assert get(1).get('experiment_accession') == 'SRX17709227'
                assert get(2).get('experiment_accession') == 'SRX17709228'
                assert get(3).get('experiment_accession') == 'SRX9315476'
            }
        }
    }
}
