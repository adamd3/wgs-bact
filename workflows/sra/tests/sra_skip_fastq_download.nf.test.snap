{
    "Parameters: --skip_fastq_download": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    "samplesheet.csv:md5,0a9b99509bcc64ca245408b6bb634f15"
                ],
                "2": [
                    "id_mappings.csv:md5,3e70c965568c59f8c8fbb8e99e7a8b79"
                ],
                "3": [
                    [
                        "multiqc_config.yml:md5,6d4c3e5137704358474330207f5f2b5c",
                        "versions.yml:md5,1496d1cbc9041e07ab8a0c25f0b054d9"
                    ]
                ],
                "4": [
                    "versions.yml:md5,1496d1cbc9041e07ab8a0c25f0b054d9",
                    "versions.yml:md5,9b17045ca8bdc272cb3f9d349a81d206",
                    "versions.yml:md5,b21338b74b9a8fe0c7a114f7686c07cd",
                    "versions.yml:md5,b52279f7d6b891a6523d9321f3f85b47"
                ],
                "fastq": [
                    
                ],
                "mappings": [
                    "id_mappings.csv:md5,3e70c965568c59f8c8fbb8e99e7a8b79"
                ],
                "sample_mappings": [
                    [
                        "multiqc_config.yml:md5,6d4c3e5137704358474330207f5f2b5c",
                        "versions.yml:md5,1496d1cbc9041e07ab8a0c25f0b054d9"
                    ]
                ],
                "samplesheet": [
                    "samplesheet.csv:md5,0a9b99509bcc64ca245408b6bb634f15"
                ],
                "versions": [
                    "versions.yml:md5,1496d1cbc9041e07ab8a0c25f0b054d9",
                    "versions.yml:md5,9b17045ca8bdc272cb3f9d349a81d206",
                    "versions.yml:md5,b21338b74b9a8fe0c7a114f7686c07cd",
                    "versions.yml:md5,b52279f7d6b891a6523d9321f3f85b47"
                ]
            }
        ],
        "timestamp": "2023-10-18T09:50:04.300075498"
    }
}